src/rust/jsg-macros/README.md

```diff
@@ -62,4 +62,44 @@ impl DnsUtil {
 }
 ```
 
-On struct definitions, generates `jsg::Type`, wrapper struct, and `ResourceTemplate` implementations. On impl blocks, scans for `#[jsg_method]` attributes and generates the `Resource` trait implementation.
+On struct definitions, generates:
+- `jsg::Type` implementation
+- `jsg::GarbageCollected` implementation (default, no-op trace)
+- Wrapper struct and `ResourceTemplate` implementations
+
+On impl blocks, scans for `#[jsg_method]` attributes and generates the `Resource` trait implementation.
+
+### Garbage Collection
+
+Resources are automatically integrated with V8's cppgc garbage collector. The macro automatically generates a `GarbageCollected` implementation that traces fields requiring GC integration:
+
+- `Ref<T>` fields - traces the underlying resource
+- `TracedReference<T>` fields - traces the JavaScript handle
+- `Option<Ref<T>>` and `Option<TracedReference<T>>` - conditionally traces
+
+```rust
+#[jsg_resource]
+pub struct MyResource {
+    // Automatically traced
+    js_callback: Option<TracedReference<Object>>,
+    child_resource: Option<Ref<ChildResource>>,
```

Comment:
What about other nested cases like `Vec` or `HashSet` of resources or references?

Do you think the automatic tracing approach could extend to these as well? Or would those always require a custom trace implementation?

Reply:
We will probably need to implement these iteratively rather than in a big chunk. We can probably get a way if we just supported Iterator trait